services/schemas/duty.zschema.ts

```diff
@@ -0,0 +1,29 @@
+import { z } from 'zod';
+import { ObjectId } from 'mongodb';
+
+const IdSchema = z.preprocess(
+  (id: any) => new ObjectId(id),
+  z.custom<ObjectId>().or(z.string()),
+);
+
+const DutySchema = z.object({
+  _id: IdSchema.optional(),
+  name: z.string(),
+  description: z.string(),
+  location: z.string(),
+  time: z.object({
+    start: z.coerce.date(),
+    end: z.coerce.date(),
+  }),
+  constraints: z.array(z.string()),
+  soldiersRequired: z.number().or(z.string()),
```

Comment:
Check and remove/comment

Reply:
resolved

Reply:
Not resolved, comment what you found

Reply:
oh I was sure I removed the `.or(z.string())` - this is what I meant

Reply:
resolved